### ISSUE TITLE
BoundaryAttack expects image as float

### ISSUE THREAD

**Issue by katywarr** [bethgelab/foolbox#297]
I'm trying to implement a boundary attack on ResNet50, but get the following error:

```
~\Anaconda3\envs\strengthening-dnns-nogpu\lib\site-packages\foolbox\attacks\boundary_attack.py in _apply_inner(self, pool, a, iterations, tune_batch_size, threaded_rnd, threaded_gen)
    194 
    195         assert self.internal_dtype in [np.float32, np.float64]
--> 196         assert external_dtype in [np.float32, np.float64]
    197 
    198         assert not (external_dtype == np.float64 and

AssertionError:
```
My initial and starting point images have integer RGB values between 0..255.
```
print(original.dtype)
print(starting_point.dtype)
```
returns
```
int32
int32
```
Here is the code that causes the error:

```
preprocessing = (np.array([104, 116, 123]), 1)
fmodel = foolbox.models.KerasModel(model, bounds=(0, 255), preprocessing=preprocessing) 
attack_criterion = foolbox.criteria.TargetClass(target_label)
attack = foolbox.attacks.BoundaryAttack(fmodel, criterion=attack_criterion)

adversarial = attack(input_or_adv = original, 
                     label = image_label,
                     starting_point = starting,
                     verbose = True)
```

I understand that the error on line 196 is because  the image dtype is expected to be a float.

My workaround is to pre-process the image (so it is a float) using the Keras provided ResNet50 pre-processing and change the bounds to (-160, 160) to cater for the updated min/max following mean normalisation. 

I.e. my code now looks like this:

```
from keras.applications.resnet50 import preprocess_input

original_for_resnet = preprocess_input(original)
starting_point_for_resnet = preprocess_input(starting_point)

fmodel = foolbox.models.KerasModel(model, bounds=(-160, 160)) 
attack_criterion = foolbox.criteria.TargetClass(target_label)
attack = foolbox.attacks.BoundaryAttack(fmodel, criterion=attack_criterion)

adversarial = attack(input_or_adv = original_for_resnet, 
                     label = image_label,
                     starting_point = starting_point_for_resnet,
                     verbose = True)
```

This fixes the problem and the attack works OK, but it doesn't feel very neat.

Have I got something wrong with the initial code?

Thanks!

**Comment by jonasrauber** [comment]
You should indeed convert your image to float, but then leave it at [0, 255] or [0, 1] and pass that as the model bounds. The preprocessing should happen either as part of the model code itself or passed to the Foolbox model class by specifying the preprocessing argument (see the example in the Readme).
